new-lamassu-admin/src/pages/Dashboard/SystemPerformance/SystemPerformance.js

```diff
@@ -0,0 +1,293 @@
+      switch (selectedRange) {
+        case 'Day':
+          if (getLastTimePeriod) {
+            return (
+              t.error === null &&
+              moment(t.created).isBetween(
+                getDateSecondsAgo(2 * 24 * 3600, now),
+                getDateSecondsAgo(24 * 3600, now)
+              )
+            )
+          }
+          return (
+            t.error === null &&
+            moment(t.created).isBetween(getDateSecondsAgo(24 * 3600, now), now)
+          )
+        case 'Week':
+          if (getLastTimePeriod) {
+            return (
+              t.error === null &&
+              moment(t.created).isBetween(
+                getDateSecondsAgo(14 * 24 * 3600, now),
+                getDateSecondsAgo(7 * 24 * 3600, now)
+              )
+            )
+          }
+          return (
+            t.error === null &&
+            moment(t.created).isBetween(
+              getDateSecondsAgo(7 * 24 * 3600, now),
+              now
+            )
+          )
+        case 'Month':
+          if (getLastTimePeriod) {
+            return (
+              t.error === null &&
+              moment(t.created).isBetween(
+                getDateSecondsAgo(60 * 24 * 3600, now),
+                getDateSecondsAgo(30 * 24 * 3600, now)
+              )
+            )
+          }
+          return (
+            t.error === null &&
+            moment(t.created).isBetween(
+              getDateSecondsAgo(30 * 24 * 3600, now),
+              now
+            )
+          )
+        default:
+          return t.error === null && true
+      }
+    }
+
+    const convertFiatToLocale = item => {
+      if (item.fiatCode === fiatLocale) return item
+      const itemRate = R.find(R.propEq('code', item.fiatCode))(data.btcRates)
+      const localeRate = R.find(R.propEq('code', fiatLocale))(data.btcRates)
+      const multiplier = localeRate.rate / itemRate.rate
+      return { ...item, fiat: parseFloat(item.fiat) * multiplier }
+    }
+
+    setTransactionsToShow(
+      R.map(convertFiatToLocale)(
+        R.filter(isInRange(false), data?.transactions ?? [])
+      )
+    )
+    setTransactionsLastTimePeriod(
+      R.map(convertFiatToLocale)(
+        R.filter(isInRange(true), data?.transactions ?? [])
+      )
+    )
+  }, [data, fiatLocale, selectedRange])
+
+  const handleSetRange = range => {
+    setSelectedRange(range)
+  }
+
+  const getNumTransactions = () => {
+    return R.length(R.filter(isNotProp('error'), transactionsToShow))
+  }
+
+  const getFiatVolume = () => {
+    // for explanation check https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/round
+    return +(
+      Math.round(
+        R.sum(getFiats(R.filter(isNotProp('error'), transactionsToShow))) +
+          'e+2'
+      ) + 'e-2'
+    )
+  }
+
+  const getProfit = (transactions = transactionsToShow) => {
+    const cashInFees = R.sum(
+      getProps('cashInFee')(R.filter(isNotProp('error'), transactions))
+    )
```

Comment:
getProps is only used here, but it's not making the intention clearer.

What about:

```
   const mapToFee = R.map(R.prop('cashInFee'))
   const noErrorTxs = filterByNoError(transactions)
   const cashInFees = mapToFee(transactions)
```